connection.go

```diff
@@ -1456,10 +1456,13 @@ func (conn *Connection) newWatcherImpl(key string, callback WatchCallback) (Watc
 				st <- state
 
 				if sendAck {
-					conn.Do(newWatchRequest(key)).Get()
 					// We expect a reconnect and re-subscribe if it fails to
 					// send the watch request. So it looks ok do not check a
-					// result.
+					// result. But we need to make sure that the re-watch
+					// request will not be finished by a small per-request
```

Comment:
What's about the timeout in initial watch request? (Or does this one implements both initial and retrying requests, it's just the deal of mildly confusing comment?)

Reply:
If the timeout is small, then the client will receive an error in case of initial watch request. To me this looks to be expected.

Reply:
Yeah, the comment is fine if this is the logic